lib/AST/CASTBridging.cpp

```diff
@@ -0,0 +1,202 @@
+#include "swift/AST/CASTBridging.h"
+
+#include "swift/AST/ASTContext.h"
+#include "swift/AST/ASTNode.h"
+#include "swift/AST/Decl.h"
+#include "swift/AST/Expr.h"
+#include "swift/AST/Stmt.h"
+#include "swift/AST/Identifier.h"
+#include "swift/AST/ParameterList.h"
+#include "swift/AST/TypeRepr.h"
+
+using namespace swift;
+
+template <typename T>
+inline llvm::ArrayRef<T> getArrayRef(BridgedArrayRef bridged) {
+  return {static_cast<const T *>(bridged.data), size_t(bridged.numElements)};
+}
+
+BridgedIdentifier
+SwiftASTContext_getIdentifier(void *ctx, const char *_Nullable str, long len) {
```

Comment:
This whole file is mind-numbingly dull to write, but at least it's straightforward!

Reply:
Yeah. I wonder if we could write a python script that gets us 90% of the way there.